### PR TITLE
Job specific ID cards

### PR DESCRIPTION
**Pull request by Alek2ander**
## About The Pull Request
Inspired by #45051

- Changes ID card sprites and adds overlays for every station job with secHUD icons and department-colored barcodes, as well as department budget card sprites with colored stickers for each department.
- The emag is also changed to fit the same shape. (unsure about this)
- Changes the wallet to fit the new ID sprite and makes the ID icon visible through it
- Improves ID card naming consistency
ids

>A nice bit of QOL I'd think.

## Changelog
:cl: Alek2ander and Triiodine
imageadd: Nanotrasen released brand-new adaptive ID cards with colorful displays. Now you can see a crew member's assignment without the need to examine any small print.
imageadd: The Syndicate immediately responded by printing their notorious cryptographic sequencer in a new form-factor.
imageadd: Wallet ID windows are now twice as big to show the new ID cards in all their glory.
/:cl:

### CODE REVIEW

**Comment by nicbn** [comment]
How can you change the ID skin ingame?

**Comment by Alek2ander** [comment]
@nicbn It's tied to the job on the ID, so any access modification console can do it. 
Making it fully customizable would be a balance change I'd rather PR separately after this is merged.

**Comment by Swiftkor** [comment]
I really like these sprites, they’re a lot more vibrant than the last pr

**Comment by Kmc2000** [comment]
Sweet

**Comment by Akrilla** [comment]
Fuck yeah

**Comment by ExcessiveUseOfCobblestone** [comment]
> Lawyer 
>> Security

**Comment by zxaber** [comment]
I wonder if you could overlay the job hud icon onto the card sprite?

The only real advantage, I suppose, would be that if new jobs are created, no one has to add custom IDs alongside everything else.

**Comment by ExcessiveUseOfCobblestone** [comment]
don't tell her anons

**Comment by Alek2ander** [comment]
@ExcessiveUseOfCobblestone Lawyers crave donuts, that's proof right there
@zxaber I'll switch to actual secHUD overlays and color-blended accents when I get around to making IDs customizable

**Comment by ExcessiveUseOfCobblestone** [comment]
They're TECHNICALLY under the jurisdiction of the HoP (Service dept?) but *shrug*

**Comment by wesoda25** [comment]
Why aren’t there silver variants for everything? Just because its silver doesn’t mean it’s always a head role. Or did you just not bother including that stuff?

**Comment by Alek2ander** [comment]
Normal, silver and gold IDs can have any job overlay.

File: code/game/objects/items/cards_ids.dm [hunk 1]
```diff
@@ -265,23 +268,41 @@
...
+		add_overlays += mutable_appearance(icon, "id[job]")
+	add_overlay(add_overlays)
+	if(istype(loc, /obj/item/storage/wallet))
+		var/obj/item/storage/wallet/powergaming = loc
+		if(powergaming.front_id == src)
+			powergaming.update_icon(add_overlays)
+
+/obj/item/card/id/get_examine_string(mob/user, thats = FALSE)
+	if(uses_overlays)
+		return "[costly_icon2html(src, user)] [thats? "That's ":""][get_examine_name(user)]" //displays all overlays in chat
```
Comment (actioninja) [inline code/game/objects/items/cards_ids.dm]: cache the result of costly_icon2html, it calls getflaticon which is very expensive

File: code/game/objects/items/storage/wallets.dm [hunk 2]
```diff
@@ -61,12 +61,20 @@
...
+		add_overlays += mutable_appearance(front_id.icon, front_id.icon_state)
+		if(override_overlays)
+			add_overlays += override_overlays
+		else
+			add_overlays += front_id.overlays
+		add_overlays += mutable_appearance(icon, "wallet_overlay")
+		add_overlay(add_overlays)
+
+/obj/item/storage/wallet/get_examine_string(mob/user, thats = FALSE)
+	return "[costly_icon2html(src, user)] [thats? "That's ":""][get_examine_name(user)]" //displays all overlays in chat
```
Comment (actioninja) [inline code/game/objects/items/storage/wallets.dm]: same for this

File: code/modules/economy/account.dm [hunk 3]
```diff
@@ -74,21 +74,21 @@
 
 			card_holder.playsound_local(get_turf(card_holder), 'sound/machines/twobeep_high.ogg', 50, TRUE)
 			if(card_holder.can_hear())
-				to_chat(card_holder, "[icon2html(A, card_holder)] *[message]*")
+				to_chat(card_holder, "[costly_icon2html(A, card_holder)] *[message]*")
```
Comment (actioninja) [inline code/modules/economy/account.dm]: _especially_ for these

File: code/modules/economy/account.dm [hunk 4]
```diff
@@ -74,21 +74,21 @@
...
 		else if(isturf(A.loc)) //If on the ground
 			for(var/mob/M in hearers(1,get_turf(A)))
 				if(M.client && !(M.client.prefs.chat_toggles & CHAT_BANKCARD) && !force)
 					return
 				playsound(A, 'sound/machines/twobeep_high.ogg', 50, TRUE)
-				A.audible_message("[icon2html(A, hearers(A))] *[message]*", null, 1)
+				A.audible_message("[costly_icon2html(A, hearers(A))] *[message]*", null, 1)
```
Comment (actioninja) [inline code/modules/economy/account.dm]: same

File: code/modules/economy/account.dm [hunk 5]
```diff
@@ -74,21 +74,21 @@
...
 				break
 		else
 			for(var/mob/M in A.loc) //If inside a container with other mobs (e.g. locker)
 				if(M.client && !(M.client.prefs.chat_toggles & CHAT_BANKCARD) && !force)
 					return
 				M.playsound_local(get_turf(M), 'sound/machines/twobeep_high.ogg', 50, TRUE)
 				if(M.can_hear())
-					to_chat(M, "[icon2html(A, M)] *[message]*")
+					to_chat(M, "[costly_icon2html(A, M)] *[message]*")
```
Comment (actioninja) [inline code/modules/economy/account.dm]: same

**Review by actioninja — CHANGES_REQUESTED** [review]

**Review by actioninja — APPROVED** [review]

**Pull request merged**